From a16fd5d9016b77d05fbe9567eac83a4be89b0de3 Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Tue, 12 Oct 2021 10:30:03 -0500
Subject: [PATCH] MGS-6424 [#ccc] Fix backwards compatibility for framebuffer

The EGL platform API has changed, breaking existing code. Update eglplatform.h
so the old API is supported as well. Fixes:

```
.../DemoFramework/FslNativeWindow/EGL/source/FslNativeWindow/EGL/FB/EGLNativeWindowSystemFB.cpp:63:25: error: invalid 'static_cast' from type 'void*' to type 'EGLNativeWindowType' {aka 'long unsigned int'}
   63 |         fbDestroyWindow(static_cast<EGLNativeWindowType>(pWindow));
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 driver/sdk/inc/EGL/eglplatform.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: imx-gpu-viv-6.4.3.p2.4-aarch32/gpu-core/usr/include/EGL/eglplatform.h
===================================================================
--- imx-gpu-viv-6.4.3.p2.4-aarch32.orig/gpu-core/usr/include/EGL/eglplatform.h
+++ imx-gpu-viv-6.4.3.p2.4-aarch32/gpu-core/usr/include/EGL/eglplatform.h
@@ -48,7 +48,7 @@
  * implementations.
  */
 
-#if defined(EGL_NO_PLATFORM_SPECIFIC_TYPES)
+#if defined(EGL_NO_PLATFORM_SPECIFIC_TYPES) || defined(EGL_API_FB)
 
 typedef void *EGLNativeDisplayType;
 typedef void *EGLNativePixmapType;
